files/restore_backup.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+import os
+import re
+import json
+import shutil
+
+DOCUMENTATION = '''
+---
+module: restore_backup
+short_description: restores a backup created with 'backup=true' using, for instance, the lineinfile module
+description:
+  - looks for a backup file in the same dir as the file to be restored and renames it to the original.
+author: Yoram Hekma
+version_added: 2.0
+requirements:
+  - python > 2.6
+options:
+  dest:
+    description:
+      - The name of the file that needs to be restored.
+    required: true
+  backup:
+    description:
+      - Create a backup file including the timestamp information so you can get
+        the original file back if you somehow clobbered it incorrectly.
+    required: false
+    choices: ["yes", "no"]
+    default: "no"
+'''
+
+EXAMPLES = '''
+  - action: restore_backup dest=/etc/samba/smb.conf
+'''
+
+
+def restore_backup(dest):
+    dirpath = os.path.dirname(dest)
+    filename = os.path.basename(dest)
+    file_dict = dict()
+    time_regexp = re.compile('^%s\.(\d{4}(:?-\d\d){2}@(:?\d\d:){2}\d\d~$)' % filename)
+
+    for f in os.listdir(dirpath):
+        # This will match: "<filename>.2015-05-26@11:50:47~" for instance
+        try:
+            timestring = re.match(time_regexp, f).group(1)
+            mtime = time.strptime(timestring, '%Y-%m-%d@%H:%M:%S~')
+            # mtime = os.stat(os.path.join(dirpath, f)).st_mtime
+            file_dict[mtime] = f
+        except AttributeError:
+            continue
+
+    mtime_list = [i for i in file_dict.keys()]
+    mtime_list.sort()
+    try:
+        latest_backup = file_dict[mtime_list[-1]]
+    except (IndexError, KeyError):
+        return {
+            'changed': False,
+            'stderr': 'No backup found',
+        }
+
+    backup_path = os.path.join(dirpath, latest_backup)
+
+    try:
+        os.rename(backup_path, dest)
+    except Exception, e:
+        return {
+            'changed': False,
+            'stderr': json.dumps("%s (renaming %s to %s)" % (e, backup_path, dest)),
```

Comment:
It looks like sometimes you get a `stderr` and some times you don't.  In general a standardized output would be good.  Also wouldn't this probably be better as a `fail_json` in this specific location?